__tests__/pypy.test.ts

```diff
@@ -0,0 +1,372 @@
+    );
+  });
+});
+
+describe('findPyPyToolCache', () => {
+  const actualPythonVersion = '3.6.17';
+  const actualPyPyVersion = '7.5.4';
+  const pypyPath = path.join('PyPy', actualPythonVersion, architecture);
+  let tcFind: jest.SpyInstance;
+  let spyReadExactPyPyVersion: jest.SpyInstance;
+
+  beforeEach(() => {
+    tcFind = jest.spyOn(tc, 'find');
+    tcFind.mockImplementation((toolname: string, pythonVersion: string) => {
+      const semverVersion = new semver.Range(pythonVersion);
+      return semver.satisfies(actualPythonVersion, semverVersion)
+        ? pypyPath
+        : '';
+    });
+
+    spyReadExactPyPyVersion = jest.spyOn(installer, 'readExactPyPyVersion');
+    spyReadExactPyPyVersion.mockImplementation(() => actualPyPyVersion);
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+    jest.resetAllMocks();
+    jest.clearAllMocks();
+  });
+
+  it('PyPy exists in the path and versions are satisfied', () => {
+    expect(finder.findPyPyToolCache('3.6.17', '7.5.4', architecture)).toEqual({
+      installDir: pypyPath,
+      resolvedPythonVersion: actualPythonVersion,
+      resolvedPyPyVersion: actualPyPyVersion
+    });
+  });
+
+  it('PyPy exists in the path and versions are satisfied with semver', () => {
+    expect(finder.findPyPyToolCache('3.6', '7.5.x', architecture)).toEqual({
+      installDir: pypyPath,
+      resolvedPythonVersion: actualPythonVersion,
+      resolvedPyPyVersion: actualPyPyVersion
+    });
+  });
+
+  it('PyPy exists in the path, but Python version is not equal', () => {
+    expect(finder.findPyPyToolCache('3.7', '7.5.4', architecture)).toEqual({
+      installDir: '',
+      resolvedPythonVersion: '',
+      resolvedPyPyVersion: ''
+    });
+  });
+
+  it('PyPy exists in the path, but PyPy version is not equal', () => {
+    expect(finder.findPyPyToolCache('3.6', '7.5.1', architecture)).toEqual({
+      installDir: null,
+      resolvedPythonVersion: '',
+      resolvedPyPyVersion: ''
+    });
+  });
+});
+
+describe('findRelease', () => {
+  const result = JSON.stringify(manifestData);
+  const releases = JSON.parse(result) as IPyPyManifestRelease[];
+  const extension = IS_WINDOWS ? '.zip' : '.tar.bz2';
+  const extensionName = IS_WINDOWS
+    ? `${process.platform}${extension}`
+    : `${process.platform}64${extension}`;
+  const files: IPyPyManifestAsset = {
+    filename: `pypy3.6-v7.3.3-${extensionName}`,
+    arch: architecture,
+    platform: process.platform,
+    download_url: `https://test.download.python.org/pypy/pypy3.6-v7.3.3-${extensionName}`
+  };
+
+  it('The specifyed python version was found, but PyPy version was not satsifyed', () => {
+    const pythonVersion = '3.6';
+    const pypyVersion = '7.3.7';
+    expect(
+      installer.findRelease(releases, pythonVersion, pypyVersion, architecture)
+    ).toEqual(null);
+  });
+
+  it('The release with python 3.6 and PyPy 7.3.3 was found', () => {
+    const pythonVersion = '3.6';
+    const pypyVersion = '7.3.3';
```

Comment:
Do we really need these variables? they are used in the single place. Let's just use them inline